### PR TITLE
Add support for the Pact smart contract language

### PR DESCRIPTION
**Pull request by emilypi**
This PR adds an entry for the [Pact smart contract language](https://www.kadena.io/pact). 

## Description

Following the directions in `CONTRIBUTORS.md`, this PR adds: 

- A language entry for Pact
- An EBNF specification for Pact
- Sample scripts used in real world applications
- Sample licenses used in real world scripts

## Checklist:
- [x] **I am adding a new language.**
  - [x] The extension of the new language is used in hundreds of repositories on GitHub.com.
    - Search results for each extension:
      
      - https://github.com/search?q=extension%3Apact+NOT+nothack&type=Code

**NB: we have many private clients that do not open source their code, but store their scripts on private enterprise github accounts, outnumbering the number of representative public samples.**

  - [x] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
      - https://github.com/kadena-io/chainweb-node/blob/master/pact/coin-contract/v3/coin-v3.pact
      - https://github.com/kadena-io/chainweb-node/blob/master/pact/coin-contract/v3/coin-v3.repl
      - https://github.com/kadena-io/pact-examples
      - https://github.com/wrappedfi/wrapped_token_pact/blob/main/wrapped-token-v1.pact

    - Sample license(s):
      - https://github.com/kadena-io/pact/blob/master/LICENSE
      - https://github.com/kadena-io/pact-lang-api/blob/master/LICENSE

  - [x] I have included a syntax highlighting grammar: https://github.com/kadena-io/pact-atom/blob/master/grammars/pact.cson
  - [x] I have updated the heuristics to distinguish my language from others using the same extension.

### CODE REVIEW

**Comment by MrMrManatee** [comment]
**I support this request as the Kadena blockchain continues to scale its operations to the masses!**

**Comment by yasinsil** [comment]
Can you please check this ?

File: lib/linguist/languages.yml
Comment (lildude) [inline lib/linguist/languages.yml:4474]: We need a sample for all extensions being added.

**Review by lildude — CHANGES_REQUESTED** [review]
Please add samples for the .repl ext or remove it from this PR.

As an aside, usage is still too low to be accepted right now.

File: lib/linguist/languages.yml
Comment (emilypi) [reply lib/linguist/languages.yml:4857]: Done

**Comment by emilypi** [comment]
@itsthejoker is there a problem you'd like to vocalize?

File: lib/linguist/languages.yml
Comment (lildude) [inline lib/linguist/languages.yml:4859]:
```suggestion
```

This field is for languages which use a shebang in some of their files. Pact doesn't appear to be one of those so we can remove this.

File: lib/linguist/languages.yml
Comment (lildude) [inline lib/linguist/languages.yml:4857]: I've looked into this a bit more closely now and there are a lot of other users of this extension which don't appear to be Pact from my naïve perspective. I'm not sure what all those other languages are but they seem to outweigh the Pact files so it's probably best to remove this extension and sample for now. Peeps can implement an override to force the language once this PR has been merged.

**Review by lildude — CHANGES_REQUESTED** [review]
Usage has indeed improved from a quick glance so we might be good to go when it comes to merging.

File: lib/linguist/languages.yml
Comment (emilypi) [reply lib/linguist/languages.yml:4857]: That sounds okay to me. There is indeed quite a bit of overlap with other languages, and I'm happy to just sit with `.pact` highlighting

Comment (emilypi) [reply lib/linguist/languages.yml]: Updated!

**Comment by emilypi** [comment]
@lildude Addressed your comments. Anything else you need from me?

**Review by jwiegley — APPROVED** [review]

File: lib/linguist/languages.yml
Comment (emilypi) [reply lib/linguist/languages.yml]: Addressed

**Comment by emilypi** [comment]
Ping!

**Comment by lildude** [comment]
Pong! 🏓

"Patience, my young Padawan". The speed of review isn't going to change the rate at which this PR will be merged as PRs are only merged close to making releases.

**Comment by lildude** [comment]
Looks good now but the language is still too low in popularity. I'll review this again around each release and will only approve the PR once the popularity requirement has been met.

**Comment by emilypi** [comment]
Alright, thanks for the update on policy so i can stop pinging 😅. We'll see if we can get more people to publish their work openly.

**Comment by XoL1507** [comment]
Eb

**Pull request merged**
